src/notebooks/R Mia Examples/Comparative-Metagenomics.qmd

```diff
@@ -0,0 +1,624 @@
+
+It's important to note, we're using methods and protocols that align with the [mia](https://microbiome.github.io/mia/) package, which can be read about further in the [OMA](https://microbiome.github.io/OMA/docs/devel/) guides.
+
+[MGnifyR](https://www.bioconductor.org/packages/release/bioc/html/MGnifyR.html) is a library that provides a set of tools for easily accessing and processing MGnify data in R, making queries to MGnify databases through the [MGnify API](https://www.ebi.ac.uk/metagenomics/api/v1/). The benefit of MGnifyR is that data can either be fetched in tsv format or be directly combined in a [TreeSummarizedExperiment](https://www.bioconductor.org/packages/release/bioc/vignettes/TreeSummarizedExperiment/inst/doc/Introduction_to_treeSummarizedExperiment.html) object to run an analysis in a custom workflow.
+
+`TreeSummarizedExperiment` is an object within the [mia bioconductor package](https://microbiome.github.io/mia/), which is dedicated to microbiome analysis.
+
+This is an interactive code notebook (a Jupyter Notebook). To run this code, click into each cell and press the ▶ button in the top toolbar, or press `shift+enter`.
+
+------------------------------------------------------------------------
+
+```{r}
+# Loading libraries
+suppressMessages({
+  library(ggplot2)
+  library(MGnifyR)
+  library(tidyverse)
+  library(vegan)
+  library(mia)
+  library(miaViz)
+  library(biomformat)
+  library(scater)
+  library(metagenomeSeq)
+  library(reshape2)
+  library(viridis)
+  library(patchwork)
+  library(IRdisplay)
+  library(ALDEx2)
+  library(ANCOMBC)
+  library(mikropml)
+  library(MLeval)
+  library(caret)
+  library(scales)
+})
+
+display_markdown(file = '../_resources/mgnifyr_help.md')
+```
+
+# Fetching data and preprocessing
+In this example we are going to fetch MGnify analysis results and metadata for a dataset on the human gut microbiome viewed across age and geography -  ([MGYS00005154](https://www.ebi.ac.uk/metagenomics/studies/MGYS00005154)).
+
+## Setting the study accession
+1. First, set the accession number of the study we're using to the `mgnify_study_accession` variable.
+
+```{r}
+#| output: false
+
+source("./utils/variable_utils.r")
+
+mgnify_study_accession <- get_variable_from_link_or_input('MGYS', 'Study Accession', 
+                                                          'MGYS00005154')
+
+# You can also just directly set the accession variable in code, like this:
+# mgnify_study_accession <- "MGYS00005292"
+```
+
+2. Then, construct a MgnifyClient object to access the database.
+
+```{r}
+#| output: false
+mg <- MgnifyClient(usecache = TRUE, cache_dir = '/home/jovyan/.mgnify_cache')
```

Comment:
MgnifyClient(useCache = TRUE, cacheDir = '/home/jovyan/.mgnify_cache')

--> they should be in camelcase (as this is constructor) https://ebi-metagenomics.github.io/MGnifyR/reference/MgnifyClient.html

Reply:
is it possible to avoid local paths?